src/MIDebugEngine/AD7.Impl/AD7Engine.cs

```diff
@@ -831,10 +854,14 @@ public int ExecuteOnThread(IDebugThread2 pThread)
         // that is, not all threads should be required to be stopped before this method returns. The implementation of this method may be 
         // as simple as calling the IDebugProgram2::CauseBreak method on this program.
         //
-        // The sample engine only supports debugging native applications and therefore only has one program per-process
         public int Stop()
         {
-            throw new NotImplementedException();
+            DebuggedProcess.WorkerThread.RunOperation(async () =>
+            {
+                await _debuggedProcess.CmdBreak();
+            });
+            // TODO: this should be returning S_ASYNC_STOP
```

Comment:
There are two problems with this code as-is:
1. I believe the engine will send an async break event. This can result in races where depending the UI could re-enter break mode after it has already continued. 
2. The engine is telling the UI that it is broken when it isn't yet - exec-interrupt doesn't necessarily synchronously break the target. But by returning S_OK we are telling the engine that we are broken. This may mean that we try to do a bunch of things that will not actually work since we aren't broken. Probably the worst of these things is that we could try to continue the target and then possibly wind up in a state where the UI thinks we are running while we are actually broken.